### DISCUSSION TITLE
scalar_one_or_none() result does not have "_mapping" or "_fields" attribures.

### DISCUSSION THREAD

**Discussion by cyberbiosecurity** [sqlalchemy/sqlalchemy#11919] [Usage Questions] [answered]
Hello.
sqlalchemy.__version__ == '2.0.34'
[Result.scalar_one_or_none](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.scalar_one_or_none) documentation chain:
1) "...This is equivalent to calling Result.scalars() **and then Result.one_or_none**()."
2) "...one_or_none() → **Row**"
3) "...class sqlalchemy.engine.**Row**
The Row object represents a row of a database result
For mapping (i.e. dictionary) behavior on a row, such as testing for containment of keys, **refer to the Row._mapping** attribute."
4) "... attribute sqlalchemy.engine.**Row._fields**

Meanwhile:
```
...
result = await session.execute(select(db.Base.classes.item_furnitures).where(db.Base.classes.item_furnitures.id == item_id))

item = result.scalar_one_or_none()
...
fields = item._fields

```

But then I see:

```
item_furnitures' object has no attribute '_fields

item_furnitures' object has no attribute '_mapping
```

What am I doing wrong in a simple operation of reading the docs and finding no attributes that are declared to exist? It's hard to believe that such widely used framework's docs are misleading in describing such widely used operation as dictionazing a row.
Thank you.

**Comment by zzzeek** [comment]
the docs say this:

   **Result.scalars()**  (call that first)  **and then** Result.one_or_none().

meaning this

    r.scalars().one_or_none()

you can try each combination at a python prompt

```
>>> from sqlalchemy import create_engine
>>> e = create_engine("sqlite://")
>>> c = e.connect()
>>> from sqlalchemy import text

>>> r = c.execute(text("select 1 as x"))
>>> row = r.one_or_none()
>>> type(row)
<class 'sqlalchemy.engine.row.Row'>
>>> row._fields
('x',)

>>> r = c.execute(text("select 1 as x"))
>>> scalar_object = r.scalars().one_or_none()
>>> type(scalar_object)
<class 'int'>
>>> scalar_object
1

>>> r = c.execute(text("select 1 as x"))
>>> scalar_object = r.scalar_one_or_none()
>>> type(scalar_object)
<class 'int'>
>>> scalar_object
1

```

**Reply by CaselIT** [reply]
I guess the doc could link `ScalarResult.one_or_none` instead of `Result.one_or_none`

**Reply by CaselIT** [reply, marked as answer]
Improved docs in 74e8e777f9aab33830d0625ef03d4a349cab24f4
